etl/readers/wikipedia_reader.py

```diff
@@ -1,26 +1,28 @@
 import json
 from collections.abc import Iterable
-from pathlib import Path
 
-from dagster import ConfigurableResource
 from unidecode import unidecode
 
-from etl.models import wikipedia
-from etl.models.types import DataFileName
+from etl.models import DataFilesConfig, wikipedia
+from etl.readers import Reader
 
 
-class WikipediaReaderResource(ConfigurableResource):  # type: ignore
-    """A ConfigurableResource to read Wikipedia data from storage."""
+class WikipediaReader(Reader):
+    """
+    A concrete implementation of Reader.
 
-    data_file_names: list[DataFileName]
+    Read in Wikipedia data and yield them as wikipedia.Articles.
+    """
 
-    def read(self) -> Iterable[wikipedia.Article]:
-        """Read in Wikipedia data and yield them as wikipedia.Article objects."""
-
-        for data_file_name in self.data_file_names:
+    def __init__(self, data_files_config: DataFilesConfig) -> None:
+        self.__parsed_data_files_config: DataFilesConfig.Parsed = (
+            data_files_config.parse()
+        )
 
-            data_file_path = Path(__file__).parent.parent / "data" / data_file_name
+    def read(self) -> Iterable[wikipedia.Article]:
+        """Read in Wikipedia data and yield them as wikipedia.Articles."""
 
+        for data_file_path in self.__parsed_data_files_config.data_file_paths:
```

Comment:
The naming should definitely be more specific here. `wikipedia_jsonl_file_paths` or similar?

Reply:
That is true, I have changed the name to `wikipedia_jsonl_file_paths`.